source/MissionPanel.cpp

```diff
@@ -355,13 +386,27 @@ bool MissionPanel::Click(int x, int y, int clicks)
 			availableIt = available.begin();
 			while(index--)
 				++availableIt;
+			availableEscortsIt = availableEscorts.end();
 			acceptedIt = accepted.end();
 			dragSide = -1;
 			selectedSystem = availableIt->Destination()->GetSystem();
 			DoScroll(available, availableIt, availableScroll, false);
 			CenterOnSystem(selectedSystem);
 			return true;
 		}
+		else if(index > available.size() && index <= available.size() + availableEscorts.size())
+		{
+			index -= available.size() + 1;
+			availableEscortsIt = availableEscorts.begin();
+			while(index--)
+				++availableEscortsIt;
+			acceptedIt = accepted.end();
+			availableIt = available.end();
+			dragSide = -1;
+			DoScroll(availableEscorts, availableEscortsIt, availableEscortsScroll, false);
+			return true;
+		}
```

Comment:
I'm a bit confused by this, why do these missions need to be handled differently from the rest; why are they not in the other available missions?
The mission has a destination by default btw, if that's what you're worried about for centering around said destination (its the source, but it could be something else and it should probably be shown here like the rest)

Reply:
Because of the extra heading I added on the job board

Reply:
Oooh right I just noticed that, good thinking.

From a player's perspective, it makes sense to show them separate for sure, but code-wise maybe we could still have them in the same vector; at the end of it, and when we display missions we do two passes, one skipping these and then another one only considering them?
I would like to have another opinion on how it should be done, I'm not sure.

Reply:
It seems like available jobs are simply sorted alphabetically, maybe they could be sorted based on if its an escort job and then alphabetically, that way we would know that once we find one escort mission the rest will be escorts too.
That said I'm still not sure that is the best way to do it

Reply:
This would be one way yes.

I will have some more thoughts on that point.